File: domain/src/main/java/com/arturogutierrez/openticator/domain/bootstrap/InitialScreenSelector.java
package com.arturogutierrez.openticator.domain.bootstrap;

import com.arturogutierrez.openticator.domain.Preferences;
import javax.inject.Inject;

public class InitialScreenSelector {

  private final Preferences preferences;

  @Inject
  public InitialScreenSelector(Preferences preferences) {
    this.preferences = preferences;
  }

  public boolean shouldShowWizard() {
    return preferences.getMasterPassword() == null;
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/bootstrap/InitialScreenSelector.kt
package com.arturogutierrez.openticator.domain.bootstrap

import com.arturogutierrez.openticator.domain.bootstrap.InitialScreenSelector.InitialScreen.LOGIN
import com.arturogutierrez.openticator.domain.bootstrap.InitialScreenSelector.InitialScreen.MAIN_SCREEN
import com.arturogutierrez.openticator.storage.preferences.Preferences
import javax.inject.Inject

class InitialScreenSelector @Inject constructor(private val preferences: Preferences) {

  enum class InitialScreen {
    LOGIN, MAIN_SCREEN
  }

  fun initialScreen(): InitialScreen {
    if (isAppConfigured()) {
      return MAIN_SCREEN
    }

    return LOGIN
  }

  private fun isAppConfigured() = preferences.masterPassword != null
}
